third_party/pcsc-lite/naclport/server_clients_management/src/client_request_processor.cc

```diff
@@ -1142,25 +1142,21 @@ void PcscLiteClientRequestProcessor::EnterConcurrencyCheckScope(
     return;
   }
 
-  std::vector<std::string> concurrent_functions;
+  std::string concurrent_functions;
   {
     std::unique_lock<std::mutex> lock(mutex_);
-    concurrent_functions.assign(currently_running_functions_.begin(),
-                                currently_running_functions_.end());
+    concurrent_functions =
+        JoinStrings(currently_running_functions_, /*separator=*/", ");
     currently_running_functions_.insert(starting_function_name);
   }
 
   if (concurrent_functions.empty()) {
     return;
   }
 
-  std::ostringstream str;
-  for (const auto& name : concurrent_functions) {
-    str << ", " << name;
-  }
   GOOGLE_SMART_CARD_LOG_WARNING
       << logging_prefix_ << "Client violates threading: concurrent calls of "
-      << starting_function_name << str.str()
+      << starting_function_name << concurrent_functions
```

Comment:
nit: We need a separator between these two since concurrent_functions no longer starts with one

Reply:
Done - good catch!